internal/transport/http_test.go

```diff
@@ -0,0 +1,141 @@
+// +build go1.16
+
+package transport_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	client "github.com/redhatinsights/yggdrasil/internal/http"
+	"github.com/redhatinsights/yggdrasil/internal/transport"
+)
+
+func startServer() {
+	getResponse := func(w http.ResponseWriter) {
+		data := map[string]string{
+			"status": "OK",
+		}
+		resBytes, _ := json.Marshal(data)
+		fmt.Fprintf(w, "%s", resBytes)
+	}
+
+	http.HandleFunc("/yggdrasil/test/401/out", func(w http.ResponseWriter, req *http.Request) {
+		w.WriteHeader(http.StatusUnauthorized)
+		getResponse(w)
+	})
+
+	http.HandleFunc("/yggdrasil/test/500/out", func(w http.ResponseWriter, req *http.Request) {
+		w.WriteHeader(http.StatusInternalServerError)
+		getResponse(w)
+	})
+
+	http.HandleFunc("/yggdrasil/test/200/out", func(w http.ResponseWriter, req *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		getResponse(w)
+	})
+
+	log.Fatalf("Cannot started mock http server: %v", http.ListenAndServe(":8889", nil))
```

Comment:
Please make sure that the server is shut down when this set of tests ends. Otherwise it will occupy chosen port until the process running tests finishes.

Reply:
Also - it might be useful to use random port; there might be port clashes on dev machines, especially around 8*** range.